server/routerlicious/packages/services-utils/src/telemetryContext.ts

```diff
@@ -50,11 +50,22 @@ function getTelemetryContextPropertiesFromRequest(
 		req.get(TelemetryContextHeaderName) ?? res.get(TelemetryContextHeaderName);
 	// Safely parse and return accepted telemetry properties.
 	const telemetryContextProperties = parseTelemetryContextHeader(telemetryContextHeader);
+	/**
+	 * Determines the source of the correlation ID based on the request headers.
+	 * If either the CorrelationIdHeaderName or TelemetryContextHeaderName is present in the request headers,
+	 * the correlation ID source is considered as "client". Otherwise, it is considered as "server".
+	 */
+	const correlationIdSource =
+		req.get(CorrelationIdHeaderName) !== undefined ||
```

Comment:
Server also sends correlationId. Remove this check, and just rely on the fact that only the server will send telemetryContextHeader. Otherwise, everything will be `client`